src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,71 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
 import './FindMovie.scss';
 
-export const FindMovie: React.FC = () => {
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieCard } from '../MovieCard';
+
+// eslint-disable-next-line max-len
+const defaultPicture = 'https://via.placeholder.com/360x270.png?text=no%20preview';
+
+type Props = {
+  onAddMovie: (newMovie: Movie) => void;
+};
+
+export const FindMovie: React.FC<Props> = ({ onAddMovie }) => {
+  const [query, setQuery] = useState('');
+  const [newMovie, setNewMovie] = useState<Movie | null>(null);
+  const [requestError, setRequestError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleChangeQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    setRequestError(false);
+  };
+
+  const handleAddMovie = () => {
+    if (newMovie) {
+      onAddMovie(newMovie);
+    }
+
+    setQuery('');
+    setNewMovie(null);
+  };
+
+  const handleSirchMovie = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setIsLoading(true);
+    getMovie(query)
+      .then(movieData => {
```

Comment:
Let’s use `async await` all over the code. It gives you a simpler way to work with asynchronous promise-based code.